worker/session.go

```diff
@@ -229,16 +230,22 @@ func (sp *sessionPool) acquire(ctx context.Context, s *session) (_ *sharedSessio
 	}
 
 reconnect:
+	ctx, cancel := context.WithTimeout(ctx, sp.sessionReconnectTimeout)
+	defer cancel()
+
 	ss.conn, err = (&net.Dialer{}).DialContext(ctx, "tcp", s.hostIP)
 	if err != nil {
+		ss.sess = nil
```

Comment:
I was on the fence on whether I should nullify the session here on failure... If we fail to reconnect I think it makes sense to do it but I wanted to make sure this did not go by unnoticed